app.py

```diff
@@ -83,22 +80,37 @@ def main():
     """Perform graph backup job."""
     _LOGGER.debug("Debug mode is on.")
 
-    _send_metrics()
+    if THOTH_DEPLOYMENT_NAME:
+        database_schema_revision_script.labels(
+            COMPONENT_NAME, GraphDatabase().get_script_alembic_version_head(), THOTH_DEPLOYMENT_NAME
+        ).inc()
+    else:
+        _LOGGER.warning("THOTH_DEPLOYMENT_NAME env variable is not set.")
 
     adapter = GraphBackupStore()
     adapter.connect()
 
     _LOGGER.info("Starting creation of the database dump")
-    run_command(
-        f"pg_dump -h {KNOWLEDGE_GRAPH_HOST} -p {KNOWLEDGE_GRAPH_PORT} "
-        f"-U {KNOWLEDGE_GRAPH_USER} -d {KNOWLEDGE_GRAPH_DATABASE} -f pg_dump.sql",
-        env={"PGPASSWORD": os.getenv("KNOWLEDGE_GRAPH_PASSWORD", "postgres")},
-        timeout=None,
-    )
-    _LOGGER.info("Uploading the database dump")
-    object_id = adapter.store_dump("pg_dump.sql")
-    _LOGGER.info("The database dump is available at %s/%s", adapter.prefix, object_id)
-    _LOGGER.info("Graph backup task is done.")
+    try:
+        run_command(
+            f"pg_dump -h {KNOWLEDGE_GRAPH_HOST} -p {KNOWLEDGE_GRAPH_PORT} "
+            f"-U {KNOWLEDGE_GRAPH_USER} -d {KNOWLEDGE_GRAPH_DATABASE} -f pg_dump.sql",
+            env={"PGPASSWORD": os.getenv("KNOWLEDGE_GRAPH_PASSWORD", "postgres")},
+            timeout=None,
+        )
+        _LOGGER.info("Uploading the database dump")
+
+        object_id = adapter.store_dump("pg_dump.sql")
+        _LOGGER.info("The database dump is available at %s/%s", adapter.prefix, object_id)
+        _LOGGER.info("Graph backup task is done.")
+
+        _METRIC_PG_DUMP_STATUS.labels(THOTH_DEPLOYMENT_NAME, __component_version__).set(0)
+
+    except Exception as e:
+        _METRIC_PG_DUMP_STATUS.labels(THOTH_DEPLOYMENT_NAME, __component_version__).set(1)
+        _LOGGER.exception(f"An error occurred while saving pg dump: {str(e)}")
```

Comment:
What about something like this (and remove `_send_metrics` bellow)?

```suggestion
        else:
            METRIC_PG_DUMP_STATUS.labels(THOTH_DEPLOYMENT_NAME, __component_version__).set(1)
        finally:
            _send_metrics()
```

It would cause the job exits with a non-zero value (and eventually restarted based on the configuration).